File: app/src/main/java/com/abhi/lokaldemo/api/model/PostResponse.java
package com.abhi.lokaldemo.api.model;

public class PostResponse {
    public String format;
    public int width;
    public int height;
    public String filename;
    public int id;
    public String author;
    public String author_url;
    public String post_url;

    public DownloadState downloadState = DownloadState.DOWNLOAD_STATE_DEFAULT;
    public int downloadProgress = 0;

    public enum  DownloadState {
        DOWNLOAD_STATE_DEFAULT,
        DOWNLOAD_STATE_PROGRESS,
        DOWNLOAD_STATE_DOWNLOADED
    }
}

File: app/src/main/java/com/abhi/lokaldemo/api/model/PostResponse.kt
package com.abhi.lokaldemo.api.model

import com.google.gson.annotations.SerializedName

data class PostResponse(
    @SerializedName("format") val format : String,
    @SerializedName("width") val width : Int,
    @SerializedName("height") val height : Int,
    @SerializedName("filename") val filename : String,
    @SerializedName("id") val id : Int,
    @SerializedName("author") val author : String,
    @SerializedName("author_url") val author_url : String,
    @SerializedName("post_url") val post_url : String
)